nemo/lightning/megatron_parallel.py

```diff
@@ -475,6 +482,37 @@ def infer_num_microbatches(self, data: Union[DataT, Iterator[DataT], List[Iterat
 
         raise ValueError("Cannot infer `num_microbatches` from data, please specify it manually")
 
+    def init_tensor_parallel_overlap(self, micro_batch_size, sequence_length):
+        import transformer_engine
+        from megatron.core import parallel_state
+
+        hidden_size = self.config.hidden_size
+        fp8 = self.config.fp8
+        ub_cfgs = self.config.tp_comm_overlap_cfg
+        cp_size = parallel_state.get_context_parallel_world_size()
+        tp_size = parallel_state.get_tensor_model_parallel_world_size()
+
+        input_shape = [
+            sequence_length * micro_batch_size // cp_size,
+            hidden_size,
+        ]
+
+        try:
+            if ub_cfgs is None:
+                logging.warning(
+                    "Tensor parallel overlap: No overlap config provided. Initializing TP comm overlap with the default config."
+                )
+            transformer_engine.pytorch.module.base.initialize_ub(
```

Comment:
If no overlap config is provided then it will default to TE's baked in configs, which may not maximize performance, so I have a warning for that here.